mypy/checkstrformat.py

```diff
@@ -151,13 +151,11 @@ def check_simple_str_interpolation(self, specifiers: List[ConversionSpecifier],
 
     def check_mapping_str_interpolation(self, specifiers: List[ConversionSpecifier],
                                        replacements: Expression) -> None:
-        if (isinstance(replacements, DictExpr) and
-                all(isinstance(k, (StrExpr, BytesExpr))
-                    for k, v in replacements.items)):
+        if isinstance(replacements, DictExpr):
```

Comment:
What about the case with a non-literal key? It might be pretty rare, but here's one example where it would be kind of reasonable:

```
# constants.py
key1 = 'key1'
key2 = 'key2'

# code.py
from constants import key1, key2
...
'%(key1)d, %(key2)d' % {key1: 1, key2: 2}
```

Not sure whether we are likely to encounter code like that in practice.